README.md

```diff
@@ -3,106 +3,214 @@
-
-For more details please proceed to [Catberry Documentation](https://github.com/catberry/catberry/blob/7.1.2/docs/index.md).
-
-Also, you can see [finished example application](https://github.com/catberry/catberry-cli/tree/master/templates/example).
+
+For more details please proceed to [Catberry Documentation](https://github.com/catberry/catberry/blob/8.0.0-dev/docs/index.md).
+
+### Typical Cat-component example
+
+```javascript
+'use strict';
+
+class CoolComponent {
+
+	/**
+	 * Creates a new instance of the "CoolComponent" component.
+	 * @param {ServiceLocator} locator The service locator for resolving dependencies.
+	 */
+	constructor(locator) {
+		// you can resolve any dependency from the locator.
+	}
+
+	/**
+	 * Gets data for the template.
+	 * This method is optional.
+	 * @returns {Promise<Object>|Object|null|undefined} Data for the template.
+	 */
+	render() {
+		return this.$context.getStoreData();
+	}
+
+	/**
+	 * Returns event binding settings for the component.
+	 * This method is optional.
+	 * @returns {Promise<Object>|Object|null|undefined} Binding settings.
+	 */
+	bind() {
+		return {
+			'a.clickable': () => window.alert('Ouch!');
+		}
+	}
+
+	/**
+	 * Cleans up everything that has NOT been set by .bind() method.
+	 * This method is optional.
+	 * @returns {Promise|undefined} Promise of nothing.
+	 */
+	unbind() {
+		// nothing to do here we have used bind properly
+	}
+}
+
+module.exports = Some;
+```
+
+The component is used as a custom tag:
+
+```html
+<cat-cool id="unique-value" cat-store="group/CoolStore"></cat-cool>
+```
+
+### Typical Store example
+
+```javascript
+'use strict';
+
+class CoolStore {
+	/**
+	 * Creates a new instance of the "CoolStore" store.
+	 * @param {ServiceLocator} locator The service locator for resolving dependencies.
+	 */
+	constructor(locator) {
+
+		/**
+		 * Current universal HTTP request for environment-independent requests.
+		 * @type {UHR}
+		 * @private
+		 */
+		this._uhr = locator.resolve('uhr');
+
+		/**
+		 * Current lifetime of data (in milliseconds) that is returned by this store.
+		 * @type {number} Lifetime in milliseconds.
+		 */
+		this.$lifetime = 60000;
+	}
+
+	/**
+	 * Loads data from a remote source.
+	 * @returns {Promise<Object>|Object|null|undefined} Loaded data.
+	 */
+	load() {
+		// Here you can do any HTTP requests using this._uhr.
+		// Please read details here https://github.com/catberry/catberry-uhr.
+	}
+
+	/**
+	 * Handles an action named "some-action" from any component or store.
+	 * @returns {Promise<Object>|Object|null|undefined} Response to the component/store.
+	 */
+	handleSomeAction() {
+		// Here you can call this.$context.changed() if you're sure'
+		// that the remote data on the server has been changed.
+		// You can additionally have many handle methods for other actions.
+	};
+}
+
+module.exports = Some;
+```
 
 ## Browser Support
-Catberry uses [ECMAScript 5](http://www.ecma-international.org/ecma-262/5.1/)
```

Comment:
While Catberry is capable of rendering pages for any browser on the server, due to the use of certain HTML5 features, like the [History API](https://developer.mozilla.org/en-US/docs/Web/Guide/API/DOM/Manipulating_the_browser_history), only partial support of old browsers is possible for the client-side javascript application.